### PR TITLE
 L1 Fee Curie: add unit test

### PR DESCRIPTION
**Pull request by DreamWuGit**
### Description

update `TxL1FeeGadgetTestContainer` to support Curie l1 fee unit tests
### Issue Link

[_link issue here_]

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by lispc** [comment]
@DreamWuGit can merge develop

**Comment by DreamWuGit** [comment]
will fix new test failure

**Comment by DreamWuGit** [comment]
@lispc  ready now.

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs [hunk 1]
```diff
@@ -407,6 +448,8 @@ mod tests {
                 is_curie,
                 gadget,
                 tx_data_gas_cost,
+                // #[cfg(feature = "l1_fee_curie")]
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs]: #[cfg(feature = "l1_fee_curie")] can be removed

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs [hunk 2]
```diff
@@ -340,61 +340,102 @@ mod tests {
...
-            TEST_FEE_SCALAR.into(),
-            TEST_TX_DATA_GAS_COST.into(),
-            TEST_TX_L1_FEE,
-        ]
-        .map(U256::from);
-
-        try_test!(TxL1FeeGadgetTestContainer<Fr>, witnesses, true);
+        // test both before & after curie upgrade
+        for is_curie in [TEST_BEFORE_CURIE, TEST_AFTER_CURIE] {
+            let witnesses = [
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:367]: the array is a bit difficult to maintain.  I would suggest a Struct

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs [hunk 3]
```diff
@@ -416,37 +459,55 @@ mod tests {
...
-                region,
-                0,
-                Value::known(witnesses[4].to_scalar().unwrap()),
-            )?;
+
+            self.is_curie
+                .assign(region, 0, Value::known(F::from(is_curie)))?;
+            let rlp_signed_len = if is_curie == 1 { tx_signed_length } else { 0 };
+            self.tx_signed_length
+                .assign(region, 0, Value::known(F::from(rlp_signed_len)))?;
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:501]: i think it is ok to still assign tx_signed_length even when is_curie == 0?    Not used anyway

Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:501]: yes

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 2 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:367]: you mean the `witness`  array , right ?

Comment (lispc) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:367]: yes

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 1 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs]: removed.

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 2 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:367]: emm, the array type is specified by the common test util `test_math_gadget_container` which this tests utilize.  even though can make structure here, but finally still need to convert to array type. 

/```
// A simple macro for less code & better readability
macro_rules! try_test {
    ($base_class:ty, $witnesses:expr, $expect_success:expr $(,)?) => {{
        test_math_gadget_container::<Fr, $base_class>($witnesses.to_vec(), $expect_success)
    }};
}
```

Comment (lispc) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:367]: i see.. then current code is also ok

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 3 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:501]: in `begin_tx` , this `tx_signed_length` is  looked up from tx table, so can not be any value eventually in real circuit even not is_curie.

**Pull request merged**